packages/venia-ui/lib/components/Dialog/dialog.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import { bool, func, shape, string, object } from 'prop-types';
+import { Form } from 'informed';
+import { X as CloseIcon } from 'react-feather';
+
+import { mergeClasses } from '@magento/venia-ui/lib/classify';
+
+import Icon from '../Icon';
+import { Modal } from '../Modal';
+import defaultClasses from './dialog.css';
+
+/**
+ * The Dialog component shows its children content in a dialog,
+ * encoding the look-and-feel and behavior in one place for consistency across the app.
+ *
+ * @typedef Dialog
+ * @kind functional component
+ *
+ * @param {Object}  props
+ * @param {BOolean} props.areButtonsDisabled - A toggle for whether the buttons are enabled.
+ * @param {Object}  props.classes - A set of class overrides to apply to elements.
+ * @param {String}  props.cancelText - The text to display on the Dialog cancel button.
+ * @param {String}  props.confirmText - The text to display on the Dialog confirm button.
+ * @param {Object}  props.formProps - Props to apply to the internal form. @see https://joepuzzo.github.io/informed/?path=/story/form--props.
+ * @param {Boolean} props.isModal - Determines behavior of clicking on the mask. False cancels Dialog.
+ * @param {Boolean} props.isOpen - Whether the Dialog is currently showing.
+ * @param {Func}    props.onCancel - A function to call when the user cancels the Dialog.
+ * @param {Func}    props.onConfirm - A function to call when the user confirms the Dialog.
+ * @param {String}  props.title - The title of the Dialog.
+ */
+const Dialog = props => {
+    const {
+        areButtonsDisabled,
+        cancelText,
+        children,
+        confirmText,
```

Comment:
It's good to be able to customize these buttons. Perhaps we should also have another prop for whether to render them at all?

```es6
const { includeButtons } = props

const buttons = includeButtons
  ? // create button elements
  : null
```

Reply:
Maybe it makes sense to have own buttons as a list with your own actions make it very changeable

Reply:
The issue with accepting any list of buttons is that this component is responsible for defining and attaching listeners to at least the standard buttons. If button customization is important, we may need a slightly more sophisticated API.

```es6
const { buttons } = props

const buttonElements = Array.from(buttons, button => {
  const { key, props } = button

  // `DialogButton` could render a `Button`,
  // attach a listener, and call an optional callback
  return <DialogButton key={key} {...props} />
})
```

Reply:
I think we decided against this.